net/src/Sails.Remoting/RemotingViaSubstrateClient.cs

```diff
@@ -73,7 +204,22 @@ public Task<byte[]> QueryAsync(
         EnsureArg.IsNotNull(programId, nameof(programId));
         EnsureArg.IsNotNull(encodedPayload, nameof(encodedPayload));
 
-        throw new NotImplementedException();
+        var nodeClient = await this.GetConnectedNodeClientAsync(cancellationToken).ConfigureAwait(false);
+
+        gasLimit ??= BlockGasLimit;
+
+        var replyInfo = await nodeClient.CalculateReplyForHandleAsync(
+                this.signingAccount.GetPublicKey(),
+                programId,
+                encodedPayload,
+                gasLimit,
+                value,
+                cancellationToken)
+            .ConfigureAwait(false);
+
+        // TODO: Check for reply code
+
+        return replyInfo.EncodedPayload;
     }
 
     private async Task<SubstrateClientExt> GetConnectedNodeClientAsync(CancellationToken cancellationToken)
```

Comment:
Better move it to another class (strategy), like `NodeProvider` with  `INodeProvider` interface and method `GetConnectedNodeClientAsync`